### PR TITLE
add page if /work/search is empty

### PR DESCRIPTION
**Pull request by kcvan**
## Description
Why did you write this code?
feat(search): add error mage if search is empty
## How to Test
- [ ] Visit /work/search
- [ ] Verify it asks you to enter a city or region and search bar shows up

Which devices did you test on?
- [x] Chrome on Mac
- [ ] Chrome on PC
- [ ] Firefox on Mac
- [ ] Firefox on PC
- [ ] Safari iPhone
- [ ] Chrome Android

### CODE REVIEW

File: src/pages/search/Search.tsx [hunk 1]
```diff
@@ -29,11 +30,38 @@ const Search = () => {
...
+              <Row className="px-4" noGutters>
+                <h5 className="text-center">Please enter in a city or region to see results.</h5>
+              </Row>
+            </Container>
+          );
+        } else if (error) {
+          return (
+            <Container tag={Fade} className={`${VIEWPORT_CENTER_LAYOUT} flex-column bee-without-header-height-container`}>
+              <h2 className="font-size-md-down-5 text-center">Error: {error.message}</h2>
+              <h5 className="text-center">Please contact support if this continues.</h5>
```
Comment (tc) [inline src/pages/search/Search.tsx]: this should have a link

File: src/pages/search/Search.tsx [hunk 2]
```diff
@@ -29,11 +30,38 @@ const Search = () => {
...
-      }
+      {({ loading, error, data }) => {
+        if (loading) {
+          return (
+            <Container tag={Fade} className={VIEWPORT_CENTER_LAYOUT}>
+              <Loading height="8rem" width="8rem" />
+            </Container>
+          );
+        }
+        if (error && error.message.includes('400') && Object.keys(queryParams).length === 0) {
```
Comment (tc) [inline src/pages/search/Search.tsx]: this is bad: `error.message.includes('400')`

Comment (tc) [reply src/pages/search/Search.tsx]: apollo errors has an `invalidArgs` parameter you can use.

Comment (tc) [reply src/pages/search/Search.tsx]: screen shot 2019-03-05 at 1 10 42 pm

i'm not sure this works either. I did an empty search and got an error instead of the search bar.

Comment (kcvan) [reply src/pages/search/Search.tsx]: ah it's because there's a 1 for guests, ill add a check for that

**Comment by kcvan** [comment]
to be used with backend PR: https://github.com/thebeetoken/beenest-backend/pull/795

File: src/pages/search/Search.tsx (hunk 1 above)
Comment (kcvan) [reply src/pages/search/Search.tsx]: 👍

**Comment by woeltjen** [comment]
When I go to `/work/search` I get:

image

Running with `kevin/ENG-1072` on the backend

File: src/pages/search/Search.tsx [hunk 3]
```diff
@@ -29,11 +31,65 @@ const Search = () => {
...
-        loading ? <Loading /> :
-        error ? <h2>Error: {error.message}</h2> :
-        <SearchPage listings={data.searchListings} />
-      }
+      {({ loading, error, data }) => {
+        if (loading) {
+          return (
+            <Container tag={Fade} className={VIEWPORT_CENTER_LAYOUT}>
+              <Loading height="8rem" width="8rem" />
+            </Container>
```
Comment (woeltjen) [inline src/pages/search/Search.tsx]: Oh yay, having the little loading thing stuck in the top-left was bugging me!

Can this be refactored into its own component, e.g. `LoadingPage`? That way we can just drop in an `if (loading) return <LoadingPage />;` wherever we GraphQL

File: src/pages/search/Search.tsx [hunk 4]
```diff
@@ -29,11 +31,65 @@ const Search = () => {
...
+              </Row>
+              <Row className="d-flex justify-content-center px-0 mx-0 bg-white bee-top">
+                <Col tag={Fade} className="w-100 p-5">
+                  <SearchBar />
+                </Col>
+              </Row>
+            </div>
+          </Container>
+          );
+        }
```
Comment (woeltjen) [inline src/pages/search/Search.tsx]: Corollary to the above, could these be their own components? I'm picturing something approximately like:

    if (loading) {
      return <LoadingPage />;
    }
    if (error) {
      return isBadUserInputError(error) ? <SearchErrorPage /> : <ErrorPage error={error} />
    }
    if (searchListings.length < 1) {
      return <EmptySearchPage />;
    }
    return <SearchPage listing={searchListings} />;

I've been leaning on this as a more general pattern when using `Query`, as deep nesting can get hard to follow.

File: src/pages/search/Search.tsx [hunk 5]
```diff
@@ -29,11 +31,65 @@ const Search = () => {
...
+
+        const { searchListings } = data;
+        if (searchListings.length === 0) {
+          return (
+            <Container tag={Fade} className="d-flex flex-column align-items-center justify-content-center bee-without-header-height-container p-0" fluid>
+            <div
+              className="bg-img-hero d-flex flex-column align-items-center justify-content-center gradient-overlay-half-primary-v1 bee-without-header-height-container w-100"
+              style={{ backgroundImage: `url('https://static.beenest.com/images/app/misc/painted-ladies2.jpg')` }}>
```
Comment (woeltjen) [inline src/pages/search/Search.tsx]: This looks almost the same as the `Jumbotron` from `Home`; would it make sense to convert these into a shared component? (Or am I missing some of the differences?)

    <Beebotron>
      <h5 className="...">No listing in this area, please try a different city or region.</h5>
    </Beebotron>

Comment (kcvan) [reply src/pages/search/Search.tsx]: they have a different height.  i agree that theyre very similar. the main thing i dont like is that we're using bootstrap/theme classes for some places and `style` for others. 

but yes, i think we can benefit from a shared component that accepts class names and styles, and src.  will make in the next PR

File: src/pages/search/Search.tsx (hunk 3 above)
Comment (kcvan) [reply src/pages/search/Search.tsx]: sounds good! i'll make a separate PR for this

File: src/pages/search/Search.tsx (hunk 4 above)
Comment (kcvan) [reply src/pages/search/Search.tsx]: great idea! will be much easier to navigate

File: src/pages/search/Search.tsx [hunk 6]
```diff
@@ -29,14 +32,75 @@ const Search = () => {
...
+      {({ loading, error, data }) => {
+        if (loading) {
+          return (
+            <Container tag={Fade} className={VIEWPORT_CENTER_LAYOUT}>
+              <Loading height="8rem" width="8rem" />
+            </Container>
+          );
+        }
+
+        if (error) isBadUserInputError(error) ? <SearchErrorPage /> : <ErrorPage error={error} />;
```
Comment (woeltjen) [inline src/pages/search/Search.tsx]: Looks like this is missing a `return`

`http://localhost:4200/work/search?locationQuery=` gives "We're sorry — something's gone wrong."

When I add a `return` statement I instead see the (really nice-looking!) "Please enter in a city or region to see results" page

File: src/pages/search/Search.tsx [hunk 7]
```diff
@@ -29,14 +32,75 @@ const Search = () => {
...
+        
+        const { searchListings } = data;
+        if (searchListings.length < 1) <EmptySearchPage />;
```
Comment (woeltjen) [inline src/pages/search/Search.tsx]: Looks like a missing `return` here as well

`http://localhost:4200/work/search?locationQuery=The+Moon` shows a page with an empty map, but when I add a `return` here I get "No listings in this area, please try a different city or region."

(Also, might be out of scope for this issue, but should we have a CTA for "Be the first to create a listing in this area"? Based on activity from the new-hosts channel it seems like searching and then creating a listing is a pretty common use case.)

File: src/pages/search/Search.tsx (hunk 6 above)
Comment (kcvan) [reply src/pages/search/Search.tsx]: 🙏

File: src/pages/search/Search.tsx (hunk 7 above)
Comment (kcvan) [reply src/pages/search/Search.tsx]: would be a good idea! our bag of nice to haves is growing😂
